src/network/networkd-link.c

```diff
@@ -854,28 +854,26 @@ static int address_handler(sd_netlink *rtnl, sd_netlink_message *m, void *userda
 static int link_push_dns_to_dhcp_server(Link *link, sd_dhcp_server *s) {
         _cleanup_free_ struct in_addr *addresses = NULL;
         size_t n_addresses = 0, n_allocated = 0;
-        char **a;
+        unsigned i;
 
         log_debug("Copying DNS server information from %s", link->ifname);
 
         if (!link->network)
                 return 0;
 
-        STRV_FOREACH(a, link->network->dns) {
-                struct in_addr ia;
+        for (i = 0; i < link->network->n_dns; i++) {
```

Comment:
I am pretty sure we should forbid this. The DNS server is the place to resolve names, and we should by design make it impossible that we create a feedback loop here and in order to resolve a name have to resolve a name...

I think it's pretty universially accepted these days that DNS servers have to be denoted as IP addresses, and I think we should do that too

(In fact we already did, we just stored the IP address as string anyway...)